src/dscanner/analysis/line_length.d

```diff
@@ -6,176 +6,141 @@
+			switch (token.value)
 			{
-				if (col > maxLineLength)
-					triggerError(tok);
-				col = 1;
+			case TOK.whitespace:
+				switch (token.ptr[0])
+				{
+				case '\t':
+					currentLineLen += 4;
+					break;
+				case '\n', '\v', '\r':
+					checkCurrentLineLength();
+					break;
+				default:
+					for (auto p = token.ptr; *p == ' '; p++)
+						currentLineLen++;
+				}
+				break;
+			case TOK.comment:
+				if (i == tokens.length - 1)
+					skipComment(to!string(token.ptr));
+				else
+					skipComment(token.ptr[0 .. tokens[i + 1].ptr - token.ptr]);
+				break;
+			case TOK.string_:
+				if (i == tokens.length - 1)
+					checkStringLiteral(to!string(token.ptr));
+				else
+					checkStringLiteral(token.ptr[0 .. tokens[i + 1].ptr - token.ptr]);
+				break;
+			default:
+				currentLineLen += token.toString().length;
 			}
-			else
-				col += getEditorLength(c);
 		}
-		return col;
 	}
 
-	unittest
+	private extern (D) void skipComment(const(char)[] commentStr)
 	{
-		assert(new LineLengthCheck(BaseAnalyzerArguments.init, 120).checkMultiLineToken(Token(tok!"stringLiteral", "		", 0, 0, 0)) == 8);
-		assert(new LineLengthCheck(BaseAnalyzerArguments.init, 120).checkMultiLineToken(Token(tok!"stringLiteral", "		\na", 0, 0, 0)) == 2);
-		assert(new LineLengthCheck(BaseAnalyzerArguments.init, 120).checkMultiLineToken(Token(tok!"stringLiteral", "		\n	", 0, 0, 0)) == 5);
-	}
-
-	static size_t tokenByteLength()(auto ref const Token tok)
-	{
-		return tok.text is null ? str(tok.type).length : tok.text.length;
-	}
-
-	unittest
-	{
-		assert(tokenByteLength(Token(tok!"stringLiteral", "aaa", 0, 0, 0)) == 3);
-		assert(tokenByteLength(Token(tok!"stringLiteral", "Дистан", 0, 0, 0)) == 12);
-		// tabs and whitespace
-		assert(tokenByteLength(Token(tok!"stringLiteral", "	", 0, 0, 0)) == 1);
-		assert(tokenByteLength(Token(tok!"stringLiteral", "    ", 0, 0, 0)) == 4);
-	}
+		import std.utf : byDchar;
 
-	// D Style defines tabs to have a width of four spaces
-	static size_t getEditorLength(C)(C c)
-	{
-		if (c == '\t')
-			return 4;
-		else
-			return 1;
+		foreach (dchar c; commentStr.byDchar)
+			if (c == '\n' || c == '\v' || c == '\r')
+				checkCurrentLineLength();
 	}
 
-	alias TokenLength = Tuple!(size_t, "length", bool, "newLine", bool, "multiLine");
-	static TokenLength tokenLength()(auto ref const Token tok, size_t prevLine)
+	private extern (D) void checkStringLiteral(const(char)[] str)
 	{
 		import std.utf : byDchar;
 
-		size_t length;
-		const newLine = tok.line > prevLine;
-		bool multiLine;
-		if (tok.text is null)
-			length += str(tok.type).length;
-		else
-			foreach (c; tok.text.byDchar)
-			{
-				if (isLineSeparator(c))
-				{
-					length = 1;
-					multiLine = true;
-				}
-				else
-					length += getEditorLength(c);
-			}
-
-		return TokenLength(length, newLine, multiLine);
+		foreach (dchar c; str.byDchar)
+		{
+			if (c == '\t')
+				currentLineLen += 4;
+			else if (c == '\n' || c == '\v' || c == '\r')
```

Comment:
why not use the existing `isLineSeparator`?

Reply:
Because of slightly different char handling logic in 3 different locations